README.md

```diff
@@ -1,29 +1,314 @@
 
-generators
-from within a rails app
 
-rails generate
-migrations
-models
-controllers
-resources
-cover the magic route resources but we'll cover that more in depth later.
-choose the smallest generate that provides you with the files you need, don't over generate, the less code in your app the better. never use scaffolds, it's too much code, learn the pattern and generate a resource.
+## Abusing Generators
 
-Don't cover scaffolds in this...
+So if generators are so amazing, why wouldn't we always want to use them all of the time? I'll answer your question with another question: a chainsaw is a wonderful tool, why wouldn't you use it for every job you have around the house? The answer to both questions is the same: great tools are only great tools if they are matched with the right task. In the same way that you should only use a chainsaw when you have a job that requires it, generators should only be used when they are needed.
+
+Extending our example from above, why wouldn't you use a chainsaw to build a model airplane?
+
+![Chainsaw Example](https://s3.amazonaws.com/flatiron-bucket/readme-lessons/chainsaw_example.jpg)
+
+In the same way as our chainsaw example, certain generators create quite a bit of code and if that code is not going to be used it will clutter the application code and cause confusion for future developers. A few years ago I took over as the lead developer for a large legacy Rails application. The previous developer had relied on generators, even when they shouldn't have been used, and the end result was that it took months to simply figure out what code was being used and what was 'garbage' code that simply came from the generators.
+
+So when is the right time to use a generator? After we've gone over the actions of each of the generators I think the answer to this query will become readily apparent and we'll walk through some case studies to help understand when each type of generator is beneficial.
+
+
+## Rails Generate
+
+Each of the generators will follow the syntax:
+
+```
+rails generate <name of generator>
+```
+
+For efficiency sake Rails created an alias method for `generate` and allows you to simply use `g`, so the CLI command above could be shortened to:
+
+```
+rails g <name of generator>
+```
+
+## Different types of generators
+
+Below are the main generators that Rails offers, we'll go through examples of each of them in their own section:
+
+* Migrations
+* Models
+* Controllers
+* Resources
+
+
+## Migration Generators
+
+Up until this point we have been creating our migrations by hand, this has been beneficial since it's important to understand how migrations work, however Rails has a great set of migration generators with conventions that can help make managing the database schema very efficient.
+
+Let's start using database migrations in our case study application and update the `posts` table. To add a new column we can use the following command:
```

Comment:
also explain the new column you want to create so the following generate command makes sense